src/events/logs.ts

```diff
@@ -0,0 +1,385 @@
+        .setThumbnail(ban.user.displayAvatarURL())
+        .addFields(
+            { name: "ID", value: ban.user.id, inline: true },
+            { name: "Created", value: `<t:${Math.floor(ban.user.createdTimestamp / millisToSecs)}>`, inline: true },
+            { name: "Bot user", value: `${ban.user.bot}`, inline: true }
+        );
+
+    return embed;
+};
+
+/**
+ * Generates an embed for whenever a user is unbanned from a server.
+ * 
+ * @param ban the ban removed
+ * @returns an embed to be logged
+ */
+export const genMemberUnbanEmbed = (ban: GuildBan): EmbedBuilder => {
+    const embed: EmbedBuilder = new EmbedBuilder()
+        .setTitle(ban.user.username)
+        .setColor(BLUE)
+        .setAuthor({
+            name: "Member banned",
+            iconURL: ban.guild.iconURL()
+        })
+        .setThumbnail(ban.user.displayAvatarURL())
+        .addFields(
+            { name: "ID", value: ban.user.id, inline: true },
+            { name: "Created", value: `<t:${Math.floor(ban.user.createdTimestamp / millisToSecs)}>`, inline: true },
+            { name: "Bot user", value: `${ban.user.bot}`, inline: true }
+        );
+
+    return embed;
+};
+
+/**
+ * Generates an embed for whenever a member's nickname or server avatar is updated.
+ * 
+ * @param before the member's state before the update
+ * @param after the member's state after the update
+ * @returns an embed for logging; or null if the member's nickname and avatar stay the same
+ */
+export const genMemberUpdateEmbed = (before: GuildMember, after: GuildMember): EmbedBuilder | null => {
+    // Properties to check
+    const oldNick: string = before.nickname;
+    const newNick: string = after.nickname;
+    const oldPfp: string = before.avatar;
+    const newPfp: string = after.avatar;
+
+    // No embed generated if nickname and avatar stay the same
+    if (
+        oldNick === newNick &&
+        oldPfp === newPfp
+    ) {
+        return null;
+    }
+
+    const embed: EmbedBuilder = new EmbedBuilder()
+        .setTitle(before.user.username)
+        .setColor(PURPLE)
+        .setAuthor({
+            name: "Server profile updated",
+            iconURL: before.guild.iconURL()
+        })
+        .setThumbnail(before.displayAvatarURL())
+        .addFields(
+            { name: "ID", value: before.id, inline: false }
+        );
+
+    // Add this if the nickname was updated
+    if (oldNick !== newNick) {
+        embed.addFields(
+            { name: "Old nickname", value: oldNick ?? before.displayName, inline: false },
+            { name: "New nickname", value: newPfp ?? after.displayName, inline: false },
+            { name: "_ _", value: "_ _", inline: true } // Empty column
+        );
+    }
+
+    // Add this if the avatar was updated
+    if (oldPfp !== newPfp) {
+        embed.addFields(
+            { name: "New avatar", value: "_ _" } // Empty field value
+        )
+            .setImage(after.displayAvatarURL());
+    }
+
+    return embed;
+};
```

Comment:
This is the embed used in the listener above
It checks for nickname and server avatar updates